tests/unit/test_schemas.py

```diff
@@ -236,23 +187,26 @@ def validateClasses(self, factory):
         for cls in protocol.getProtocolClasses():
             instance = factory(cls)
             jsonDict = instance.toJsonDict()
-            self.assertTrue(cls.validate(jsonDict))
+            if not cls.validate(jsonDict):
+                print(cls.__name__)
+                print(avrotools.ValidationTool.getInvalidFields(
+                    cls, jsonDict))
+                self.fail("validation failed")
 
-    @unittest.skipIf(protocol.version.startswith("0.6"), "")
     def testValidateDefaultValues(self):
         self.validateClasses(self.getDefaultInstance)
 
-    @unittest.skipIf(protocol.version.startswith("0.6"), "")
     def testValidateTypicalValues(self):
-        self.validateClasses(self.getTypicalInstance)
+        self.validateClasses(avrotools.CreationTool.getInstance)
 
+    # TODO fix ReadAlignment.alignments decode type ambiguity!
```

Comment:
Crud; this is a nasty one. We'll have to think about this... The `embeddedTypes` short cut was always a nasty hack, and I guess it's after catching up with us now.